src/modules/ethereum/tradeTransactions.js

```diff
@@ -0,0 +1,124 @@
+import store from "modules/store";
+import ethers from "ethers";
+import moment from "moment";
+import BigNumber from "bignumber.js";
+import { ONE_ETH_IN_WEI } from "utils/constants";
+
+export async function fetchTradesTx(account, fromBlock, toBlock) {
+    try {
+        const exchangeInstance = store.getState().exchange.contract.ethersInstance;
+        const NewOrder = exchangeInstance.interface.events.NewOrder;
+        const OrderFill = exchangeInstance.interface.events.OrderFill;
+        const CancelledOrder = exchangeInstance.interface.events.CancelledOrder;
+        const provider = store.getState().web3Connect.ethers.provider;
+
+        const paddedAccount = ethers.utils.hexlify(ethers.utils.padZeros(account, 32));
+        const [logsNewOrder, logsOrderFillBuy, logsOrderFillSell, logsCanceledOrder] = await Promise.all([
+            provider.getLogs({
+                fromBlock: fromBlock,
+                toBlock: toBlock,
+                address: exchangeInstance.address,
+                topics: [NewOrder.topics[0], null, paddedAccount]
+            }),
+            provider.getLogs({
+                fromBlock: fromBlock,
+                toBlock: toBlock,
+                address: exchangeInstance.address,
+                topics: [OrderFill.topics[0], paddedAccount]
+            }),
+            provider.getLogs({
+                fromBlock: fromBlock,
+                toBlock: toBlock,
+                address: exchangeInstance.address,
+                topics: [OrderFill.topics[0], null, paddedAccount]
+            }),
+            provider.getLogs({
+                fromBlock: fromBlock,
+                toBlock: toBlock,
+                address: exchangeInstance.address,
+                topics: [CancelledOrder.topics[0], null, paddedAccount]
+            })
+        ]);
+        const logs = await Promise.all([
+            ...logsNewOrder.map(eventLog => _formatTradeLog(NewOrder, account, eventLog)),
+            ...logsCanceledOrder.map(eventLog => _formatTradeLog(CancelledOrder, account, eventLog)),
+            ...logsOrderFillBuy.map(eventLog => _formatTradeLog(OrderFill, account, eventLog, 'buy')),
+            ...logsOrderFillSell.map(eventLog => _formatTradeLog(OrderFill, account, eventLog, 'sell'))
+        ]);
+
+        logs.sort((log1, log2) => {
+          return log2.blockData.timestamp - log1.blockData.timestamp;
```

Comment:
minor improvements:

1. use blockNumber for sorting (as for transfers so we don't rely on block timestamp being fetched)
1. maybe do the sort in reducer (as for transfers so it's implemented the same way and custom sorting could be passed later to the reducer ?)